### PR TITLE
Upload a file to created opportunity Lightning UE

### PR DESCRIPTION
**Pull request by JuanCarlosMartinez4**

### CODE REVIEW

File: src/main/java/salesforce/ui/pages/notesattachments/NotesAndAttachmentsClassicPage.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+/**
+ * Manages notes and attachments on Classic user experience.
+ *
+ * @author Juan Martinez.
+ * @version 1.0 21 March 2020.
+ */
+public class NotesAndAttachmentsClassicPage extends NotesAndAttachmentsPageAbstract {
+
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/notesattachments/NotesAndAttachmentsClassicPage.java]: deletes this line

File: src/main/java/salesforce/ui/pages/notesattachments/uploadfiles/UploadFilesPopUp.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import salesforce.ui.pages.BasePage;
+
+/**
+ * Allows to show upload files popup.
+ *
+ * @author Juan Martinez.
+ * @version 1.0 23 March 2020.
+ */
+public class UploadFilesPopUp extends BasePage {
+
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/notesattachments/uploadfiles/UploadFilesPopUp.java]: deletes this line

File: src/main/java/salesforce/ui/pages/opportunities/OpportunitiesPageAbstract.java [hunk 3]
```diff
@@ -20,11 +20,6 @@
  */
 public abstract class OpportunitiesPageAbstract extends BasePage {
 
-    @Override
-    protected void waitUntilPageObjectIsLoaded() {
-
-    }
-
     /**
      * Display a opportunity list.
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/opportunities/OpportunitiesPageAbstract.java]: an opportunity

File: src/main/java/salesforce/ui/pages/opportunities/OpportunitiesPageLightning.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import salesforce.ui.pages.opportunity.OpportunityPageAbstract;
+
+/**
+ * Manages opportunities page lightning.
+ *
+ * @author Juan Martinez.
+ * @version 1.0 23 March 2020.
+ */
+public class OpportunitiesPageLightning extends OpportunitiesPageAbstract {
+
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/opportunities/OpportunitiesPageLightning.java]: deletes this line

File: src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java [hunk 5]
```diff
@@ -12,11 +12,16 @@
 /**
  * Defines an OpportunityClassicPage.
  *
- * @author Enrique Carrizales.
- * @version 1.0 21 March 2020.
+ * @author Juan Martinez.
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java:24]: copyright! Enrrique Carrizales

File: src/main/java/salesforce/ui/pages/opportunity/OpportunityLightningPage.java [hunk 6]
```diff
@@ -9,28 +9,49 @@
...
 import org.openqa.selenium.support.FindBy;
 import org.openqa.selenium.support.ui.ExpectedConditions;
 
 /**
- * Defines an OpportunityLightningPage.
+ * Manages opportunity page lightning.
  *
- * @author Enrique Carrizales.
- * @version 1.0 21 March 2020.
+ * @author Juan Martinez.
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/opportunity/OpportunityLightningPage.java:22]: Copyright! Enrrique Carrizales

File: src/main/java/salesforce/ui/pages/opportunity/OpportunityLightningPage.java [hunk 7]
```diff
@@ -9,28 +9,49 @@
...
+ * @version 1.0 23 March 2020.
  */
 public class OpportunityLightningPage extends OpportunityPageAbstract {
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/opportunity/OpportunityLightningPage.java:49]: deletes this line

File: src/test/java/salesforcetest/steps/OpportunityStep.java [hunk 8]
```diff
@@ -103,7 +103,7 @@ public void deleteOpportunities() {
     /**
      * Navigates to an opportunity.
      */
-    @And("I navigate to Opportunities Page")
+    @When("I navigate to Opportunities Page")
     public void navigatesToOpportunitiesPage() {
         PageTransporter pageTransporter = new PageTransporter();
         pageTransporter.navigateToPage(OPPORTUNITY_PAGE);
```
Comment (Amet015) [inline src/test/java/salesforcetest/steps/OpportunityStep.java]: here you need to send the page throug the scenario

Comment (JuanCarlosMartinez4) [reply src/test/java/salesforcetest/steps/OpportunityStep.java]: in this step I only change the decorator **@When**

File: src/main/java/salesforce/ui/pages/notesattachments/NotesAndAttachmentsPageAbstract.java [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+     */
+    public abstract void clickOnUploadFiles(String filePath);
+
+    /**
+     * Gets uploaded file name.
+     * @param fileName value.
+     * @return uploaded file name.
+     */
+    public abstract String getUploadedFileName(String fileName);
+}
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/notesattachments/NotesAndAttachmentsPageAbstract.java:34]: ctrl + alt + L please

File: src/main/java/salesforce/ui/pages/notesattachments/NotesAndAttachmentsLightningPage.java [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+    @FindBy(xpath = "//a[@title='Upload Files']")
+    private WebElement uploadFilesButton;
+
+    /**
+     * Gets a web element by xpath.
+     * @param xpathValue string.
+     * @param textToConcat value.
+     * @return a web element.
+     */
+    private WebElement getWebElement(final String xpathValue, final String textToConcat) {
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/notesattachments/NotesAndAttachmentsLightningPage.java:42]: ctrl + alt + L please

File: src/main/java/salesforce/ui/pages/notesattachments/NotesAndAttachmentsLightningPage.java [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+        clickOnUploadFilesButton();
+        UploadFileRobot.uploadFile(fileAbsolutePath);
+        closeUploadFilesPopUp();
+    }
+
+    /**
+     * Gets uploaded file name.
+     * @param fileName param.
+     * @return uploaded file name.
+     */
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/notesattachments/NotesAndAttachmentsLightningPage.java:80]: ctrl + alt + L please

File: src/main/java/salesforce/ui/pages/opportunities/AbstractOpportunitiesPage.java [hunk 12]
```diff
@@ -40,5 +35,12 @@ protected void waitUntilPageObjectIsLoaded() {
      * @param opportunityName value.
      * @return OpportunityPage instance.
      */
+    public abstract AbstractOpportunityPage clickOnOpportunityName(String opportunityName);
+
+    /**
+     * Selects opportunity by name.
+     * @param opportunityName value.
+     * @return opportunity page.
+     */
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/opportunities/AbstractOpportunitiesPage.java:44]: ctrl + alt + L please

File: src/main/java/salesforce/ui/pages/opportunities/OpportunitiesClassicPage.java [hunk 13]
```diff
@@ -27,6 +28,22 @@
...
+
+    @FindBy(xpath = "//table[@class='list']")
+    private WebElement opportunityTable;
+
+    /**
+     * Allows to get and web element.
+     * @param xpathValue string.
+     * @param textToConcat value.
+     * @return a web element.
+     */
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/opportunities/OpportunitiesClassicPage.java:42]: ctrl + alt + L please

File: src/main/java/salesforce/ui/pages/opportunities/OpportunitiesLightningPage.java [hunk 14]
```diff
@@ -49,12 +52,40 @@ public void displayOpportunityList(final String listName) {
         webDriver.findElement(By.xpath(opportunityOrderedListLocator)).click();
     }
 
-    @Override
+    /**
+     * Selects opportunity name.
+     * @param opportunityName value.
+     * @return opportunity page.
+     */
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/opportunities/OpportunitiesLightningPage.java:59]: ctrl + alt + L please

File: src/main/java/salesforce/ui/pages/opportunities/OpportunitiesLightningPage.java [hunk 15]
```diff
@@ -49,12 +52,40 @@ public void displayOpportunityList(final String listName) {
...
+     * @return a web element.
+     */
+    private WebElement getWebElement(final String xpathValue, final String textToConcat) {
+        return webDriver.findElement(By.xpath(String.format(xpathValue, textToConcat)));
+    }
+
+    /**
+     * Clicks on opportunity link name.
+     * @param opportunityName value.
+     */
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/opportunities/OpportunitiesLightningPage.java:81]: ctrl + alt + L please

**Pull request closed without merging**